tests/unit/modules/test_check.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import pytest
+
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    errors, utils,
+)
+from ansible_collections.sensu.sensu_go.plugins.modules import check
+
+from .common.utils import (
+    AnsibleExitJson, AnsibleFailJson, ModuleTestCase, set_module_args,
+)
+
+
+class TestSensuGoCheck(ModuleTestCase):
+    def test_minimal_check_parameters(self, mocker):
+        sync_mock = mocker.patch.object(utils, "sync")
+        sync_mock.return_value = True, {}
+        set_module_args(
+            name="test_check",
+            command='echo "test"',
+            subscriptions=['switches'],
+            interval=60
+        )
+
+        with pytest.raises(AnsibleExitJson) as context:
+            check.main()
+
+        state, _client, path, payload, check_mode = sync_mock.call_args[0]
+        assert state == "present"
+        assert path == "/checks/test_check"
+        assert payload == dict(
+            command='echo "test"',
+            subscriptions=['switches'],
+            interval=60,
+            timeout=30,
+            publish=False,
+            handlers=[],
+            env_vars=None,
+            output_metric_handlers=None,
+            check_hooks=None,
+            subdue=None,
+            ttl=0,
+            output_metric_format='',
+            proxy_entity_name='',
+            high_flap_threshold=0,
+            low_flap_threshold=0,
+            round_robin=False,
+            stdin=False,
+            runtime_assets=None,
+            metadata=dict(
+                name="test_check",
+                namespace="default",
+            ),
+        )
```

Comment:
The payload above should really be:

```python
dict(
  command='echo "test"',
  subscriptions=['switches'],
  interval=60,
  metadata=dict(
    name="test_check",
    namespace="default",
  )
)
```

All other default values are wrong - they should not be part of the payload. Please do following:

- update module args specification so that it doesn't set default values for any of optional parameters
- update the code so that it doesn't include the parameter in payload when the value is None